xrally_kubernetes/env/platforms/existing.py

```diff
@@ -129,3 +129,98 @@ def _get_validation_context(self):
     def info(self):
         version = k8s_service.Kubernetes(self.platform_data).get_version()
         return {"info": version}
+
+    @classmethod
+    def create_spec_from_sys_environ(cls, sys_environ):
+        """Create a spec based on system environment.
+
+        The environment variables could be defined with two mutually exclusive
+        mandatory ways: defining certificates or defining auth token.
+
+        To define certificates to connect next variables used:
+        .. envvar:: KUBERNETES_HOST
+            The URL to the Kubernetes host.
+        .. envvar:: KUBERNETES_TLS_INSECURE
+            Not to verify the host against a CA certificate.
+        .. envvar:: KUBERNETES_CERT_AUTH
+            A path to a file containing TLS certificate to use when
+            connecting to the Kubernetes host.
+        .. envvar:: KUBERNETES_CLIENT_CERT
+            A path to a file containing client certificate to use when
+            connecting to the Kubernetes host.
+        .. envvar:: KUBERNETES_CLIENT_KEY
+            A path to a file containing client key to use when connecting to
+            the Kubernetes host.
+
+        To define auth token to connect next variables used:
+        .. envvar:: KUBERNETES_HOST
+            The URL to the Kubernetes host.
+        .. envvar:: KUBERNETES_CERT_AUTH
+            A path to a file containing TLS certificate to use when
+            connecting to the Kubernetes host.
+        .. envvar:: KUBERNETES_API_KEY
+            Client API key to use as token when connecting to the Kubernetes
+            host.
+        .. envvar:: KUBERNETES_API_KEY_PREFIX
+            Client API key prefix to use in token when connecting to the
+            Kubernetes host.
+       """
```

Comment:
The information of the plugin which is displayed at plugins reference page is generated based on the return value of [get_info method](https://github.com/openstack/rally/blob/master/rally/common/plugin/info.py#L125). This method uses by default class.__doc__ string, but we can override this behavior to include documentation of environment variables.

Override https://github.com/openstack/rally/blob/master/rally/common/plugin/info.py#L116 method to look something like:

```python

def _get_doc(cls):
    doc = trim(cls.__doc__)
    doc += "\n **Create a spec based on system environment.**\n"
    # cut the first line since we already included the first line of it.
    doc += trim(cls.create_spec_from_sys_environ.__doc__)[1:]
    return doc
```
To ensure that this works you can call ``rally plugin show existing@kubernetes``